### ISSUE TITLE
Potential issue with function Pst.from_io_files()

### ISSUE THREAD

**Issue by flydream0428** [pypest/pyemu#382]
I am trying to use `Pst.from_io_fiels()` to initiate a PST class. If I run the PEST INSCHEK utility, I don't receive any error. However, pyEmu will complain about this. 

My code:
==================================
```
tpl_files = ['pp_brooks.pts.tpl', 
            'pp_ddftr.pts.tpl',
            'pp_kenm.pts.tpl',
            'pp_kx.pts.tpl',
            'pp_ss.pts.tpl',
            'pp_ssim_mul.pts.tpl',
            'pp_sy.pts.tpl',
            'pp_vka.pts.tpl',
            'scalar_par.dat.tpl']
in_files = [r'preproc\pp_brooks.pts', 
            r'preproc\pp_ddftr.pts',
            r'preproc\pp_kenm.pts',
            r'preproc\pp_kx.pts',
            r'preproc\pp_ss.pts',
            r'preproc\pp_ssim_mul.pts',
            r'preproc\pp_sy.pts',
            r'preproc\pp_vka.pts',
            r'preproc\scalar_par.dat']
ins_files = ["hd_obs_model_diff.smp.ins",           
            'wp_obs_model.dat.ins',
            'sub_obs_model.smp.ins']
out_files = [r"postproc\hd_obs_model_diff.smp",        
            r'postproc\wp_obs_model.dat',
            r'postproc\sub_obs_model.smp']
# ins_files = ["my.ins"]
# out_files = ["my.out"]
pst = pyemu.Pst.from_io_files(tpl_files, in_files, ins_files, out_files)
```

Errors:
===================================
```
error using inschek for instruction file sub_obs_model.smp.ins:run() returned non-zero: 1
observations in this instruction file will havegeneric values.
noptmax:0, npar_adj:8361, nnz_obs:48837
```

Potential reason:
==================================
It seems like it is related to the column length specified in the instruction file. 
For example, for the two lines below in a model output file, 
obs1        7.1919418E-04
obs2         3.720587 
Because the length is not fixed for the output values, we want to use a wide column here. 
However, with this wide column, for obs2, it has exceeded the length of the line. However, PEST doesn't complain about this. 
I guess it will ignore anything beyond the length of the line. I also notice that I have to use () rather than [] for PESTPP-SWP, while other PESSPP and PEST also work for [].
l1 (obs1)10:26
l1 (obs2)10:26

**Comment by jtwhite79** [comment]
Hey Tao - what you are seeing is a warning that the system call to `inschek`returned non-zero, meaning either it wasnt found (in the path) or it didnt like something...

what version of pyemu are you using?  The more recent versions have built-in instruction file processing that usually (well, sometimes) gives some clues about what went wrong with trying to read the existing output file with instruction file...

Also, all of the PEST++ tools use the same instruction file handlers under the hood so they should be consistent.  Can you check which PEST++ version also?  

For that style of output file, you could also use "free-format" style like (which can be a little safer if the observation names get longer and the floating point format is changing wildly...):

pif ~
l1 !dum! w !obs1!
l1 !dum! w !obs2!

**Comment by briochh** [comment]
You've probably already been there but worth checking the lengths of the two files and the line continuation commands. Could be that we are the instruction is misaligned with the output file and therefore we are looking for and obs that isn't there (and pyemu will just dump in that generic value)

**Comment by flydream0428** [comment]
Thanks, Jeremy and @briochh , I have tested a few things. 

Essentially this is because I have a model output file with a non-fixed line width. If I use the free-format style. Everything works. 
Meanwhile, I also noticed something I could not figure out when the semi-fixed format was used. 

1. pyemu version, it returns ''0+unknown'' when I call `pyemu.__version__`. I also try to reinstall it using `pip install pyemu`. I get 
![image](https://user-images.githubusercontent.com/15154177/202333986-e4c2fc69-c820-4144-9394-e5e5845f7738.png). I guess I have the latest version. 

2. It is true there is a problem with running 'inschek' on my machine due to IT's security setup. I have fixed this. With this fix, I now receive a different error code. 
![image](https://user-images.githubusercontent.com/15154177/202349355-d89a51f7-2a2a-4266-809e-668150278a0f.png)
I can run 'inschek' manually in the same folder. I can also run it with the following pyemu codes. However, it will fail when pyemu call the same line through 'Pst.from_io_fiels()'
`ins_file = 'sub_obs_model.smp.ins'
out_file = 'postproc/sub_obs_model.smp'
cwd='.'
pyemu.os_utils.run("inschek {0} {1}".format(ins_file, out_file), cwd=cwd)
obf_file = os.path.join(cwd, ins_file.replace(".ins", ".obf"))
df = pd.read_csv(obf_file, delim_whitespace=True, skiprows=0, index_col=0, names=["obsval"])`

3. PESTPP version. I am using PESTPP 5.1.24 The PESTPP-GLM I tested is an old version 4.2.16. 

To summarise, with a free-format ins file, we can solve this : )

**Comment by briochh** [comment]
Hi @flydream0428, happy for me to close this. Seems you got it sorted with free-format ins file? Reopen if still an issue.
